### PR TITLE
Change FBs of powerlink to be compatible with latest FORTE version

### PR DESCRIPTION
**Pull request by gafert**
This pull request fixes and adds features in the powerlink module fixing #169: 
* Removes the old FBs used by powerlink and adds new ones in the new structure
* Fixes incorrectly named variables introduced in https://github.com/eclipse-4diac/4diac-forte/commit/3cd52d7c9d66c7923f6f1f2973bbec64b91f3d2a
* Adds automatic download and compilation of openPOWERLINK and tinyXML in CMake with FetchContent
* Adds powerlink in systemtests

Compile it with `-DFORTE_ARCHITECTURE=Posix -DFORTE_COM_ETH=ON -DFORTE_MODULE_POWERLINK=ON`

(Problem with OPC UA which was described here resolved in [e87b059](https://github.com/eclipse-4diac/4diac-forte/pull/171/commits/e87b0593ab1eb239eea6a870b573b36f3ad1775f))

### CODE REVIEW

**Comment by azoitl** [comment]
@gafert in order that we can accept your contribution you need to link your Github account with your eclipse account. Use the same email address for your commits as the one used for your Eclipse account, and sign the Eclipse Contributor Agreement.

**Comment by gafert** [comment]
I tested it now for a few weeks and it seems to be stable. 

@azoitl I saw your [comment ](https://github.com/eclipse-4diac/4diac-forte/issues/169#issuecomment-2186396050) and I agree. For now I just wanted to make it work again.

**Comment by azoitl** [comment]
@gafert please ensure that all files have a correct copy right header. Especially the files in the FB directory don't have that. The header generated by 4diac IDE 4diac FORTE export is not suitable for this repo here.

**Comment by gafert** [comment]
@azoitl I updated to copyright. Does that work for you?

File: src/com/powerlink/fb/POWERLINK_MN_fbt.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+* Copyright (c) 2012 - 2014 AIT
```
Comment (azoitl) [inline src/com/powerlink/fb/POWERLINK_MN_fbt.h]: I guess this should be 2024

**Review by azoitl — CHANGES_REQUESTED** [review]
several method parameters, struct and class names now violate our Codeing rules. can you please fix that.

**Comment by gafert** [comment]
@azoitl I fixed the coding style according to the PDF. I also updated to the new FB style you just introduced. Please tell me if there is anything else or I have missed something :)

File: src/com/powerlink/EplWrapper.h [hunk 2]
```diff
@@ -18,90 +18,93 @@
...
-        delete *(TEplMappingList::Iterator) mCurrentValues.begin();
-        mCurrentValues.popFront();
+struct EplMapping {
+  struct EplMappingValues {
+    unsigned int mDataSize;
+    unsigned int mPiOffset;
+    unsigned int mBitOffset;
+    char *mCurrentValue;
+
+    EplMappingValues(unsigned int pa_nDataSize, unsigned int pa_nPiOffset,
```
Comment (azoitl) [inline src/com/powerlink/EplWrapper.h]: we are not useing the type indicators anymore and where possible we are removing them. Therefore please don't add them anymore.

File: src/com/powerlink/EplWrapper.h [hunk 3]
```diff
@@ -18,90 +18,93 @@
...
+        mCurrentValue[i] = 0x00;
       }
     }
+
+    ~EplMappingValues() {
+      delete[] mCurrentValue;
+    }
+
+  private:
+    EplMappingValues(const EplMappingValues &obj);
```
Comment (azoitl) [inline src/com/powerlink/EplWrapper.h]: should be `paObj`

File: src/com/powerlink/EplWrapper.h (hunk 2 above)
Comment (gafert) [reply src/com/powerlink/EplWrapper.h]: I have removed the type indicators

**Review by azoitl — APPROVED** [review]

**Pull request merged**